clients/common/nm-secret-agent-simple.c

```diff
@@ -443,6 +460,290 @@ add_vpn_secrets (NMSecretAgentSimpleRequest *request,
 	return TRUE;
 }
 
+typedef struct {
+	GPid auth_dialog_pid;
+	char read_buf[5];
+	GString *auth_dialog_response;
+	NMSecretAgentSimpleRequest *request;
+	GPtrArray *secrets;
+	guint child_watch_id;
+	gulong cancellable_id;
+} AuthDialogData;
+
+static void
+_auth_dialog_data_free (AuthDialogData *data)
+{
+	g_ptr_array_unref (data->secrets);
+	g_spawn_close_pid (data->auth_dialog_pid);
```

Comment:
this is not needed on Posix systems. Here the pid is a plain integer, not a handle. We don't need to pretend to run on anything but Linux.

Reply:
"It should be used on all platforms" -- glib Manual

That it doesn't do anything is an implementation detail of the current version. A future version of GLib may choose to change it to do something.